conda_libmamba_solver/shards.py

```diff
@@ -0,0 +1,341 @@
+# Copyright (C) 2022 Anaconda, Inc
+# Copyright (C) 2023 conda
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+Models for sharded repodata, and to make monolithic repodata look like sharded
+repodata.
+"""
+
+from __future__ import annotations
+
+import concurrent.futures
+import time
+from collections import defaultdict
+from typing import TYPE_CHECKING, TypedDict
+from urllib.parse import urljoin
+
+import conda.gateways.repodata as repodata
+import msgpack
+import zstandard
+from conda.base.context import context
+from conda.gateways.connection.session import get_session
+from conda.gateways.repodata import (
+    _add_http_value_to_dict,
+    conda_http_errors,
+)
+from conda.models.records import PackageRecord
+from requests import HTTPError
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from conda.core.subdir_data import SubdirData
+    from conda.gateways.repodata import (
+        RepodataCache,
+    )
+    from requests import Response
+
+
+Shard = TypedDict("Shard", {"packages": dict[str, dict], "packages.conda": dict[str, dict]})
+
+
+class RepodataInfo(TypedDict):  # noqa: F811
+    base_url: str  # where packages are stored
+    shards_base_url: str  # where shards are stored
+    subdir: str
```

Comment:
Can we pick one style of defining the TypedDicts for consistency? I like the `class` version better.